src/views/List.jsx

```diff
@@ -20,28 +22,76 @@ export function List({ data, loading }) {
 		return shoppingListArr.length === 0 ? true : false;
 	};
 
+	const handleClick = (e) => {
+		e.preventDefault();
+		setFilterInput('');
+	};
+
+	/*Handles navigation to Add Item view */
+	const handleAddItem = () => {
+		navigate('/add-item');
+	};
+
+	//Function to assign string value to buyingUrgency and color value to colorUrgency
+	const getBuyingUrgency = (item) => {
+		//Returns the difference between the currentDate and dateNextPurchased
+		let buyingUrgency;
+		let colorUrgency;
+
+		if (item.dateNextPurchased) {
+			let daysUntilNextPurchase = getDaysBetweenDates(
+				new Date(),
+				item.dateNextPurchased.toDate(),
+			);
+
+			//Conditional statement to categorize if an item is:
+			// - inactive: (60 days have passed since the last purchase)
+			// - overdue: currentDate has passed the dateNextPurchased, but not yet inactive
+			// - not soon: (30 days or more until the next purchase)
+			// - kind of soon: (between 7 & 30 days until the next purchase)
+			// - soon: (7 days or fewer until the next purchase)
+			if (daysUntilNextPurchase >= 60) {
+				buyingUrgency = 'inactive';
+				colorUrgency = '#878E88';
+			} else if (new Date() > item.dateNextPurchased.toDate()) {
+				buyingUrgency = 'overdue';
+				colorUrgency = '#A30000';
+			} else if (daysUntilNextPurchase >= 30) {
+				buyingUrgency = 'not soon';
+				colorUrgency = '#004777';
+			} else if (daysUntilNextPurchase > 7 && daysUntilNextPurchase < 30) {
+				buyingUrgency = 'kind of soon';
+				colorUrgency = '#00AFB5';
+			} else {
+				buyingUrgency = 'soon';
+				colorUrgency = '#FF7700';
+			}
+		}
+
+		return { buyingUrgency, colorUrgency };
+	};
+
+	/* Sorting the shopping list items by urgency */
+	let dataWithUrgency = data
+		.map((item) => ({
+			...item,
+			urgency: getBuyingUrgency(item),
+		}))
+		.filter((item) => item.hasOwnProperty('name'))
+		.sort(comparePurchaseUrgency);
```

Comment:
this is 100% something to consider in the future - it might be a little too clever to be worthwhile, but you may be able to use the calculated `buyingUrgency` property combined with an appropriate date property to do the sorting.